src/base/emu-i386/simx86/codegen-x86.c

```diff
@@ -1945,19 +1946,19 @@ arith0:		{
 		    }
 		    if (mode&MOVSSRC) {
 			// esi = base1 + CPU_(e)SI +- n
+			// subl %%[re]bp,%%[re]si
+			Gen48(Cp); G2M(0x29,0xee,Cp);
 			// subl OVERR_DS(%%ebx),%%esi
 			G2(0x732b,Cp); G1(IG->ovds,Cp);
-			// subl MEMBASE(%%ebx),%%esi
-			G3M(0x2b,0x73,Ofs_MEMBASE,Cp);
 			// movw %%si,Ofs_SI(%%ebx)
 			G4M(0x66,0x89,0x73,Ofs_SI,Cp);
 		    }
 		    if (mode&MOVSDST) {
 			// edi = base2 + CPU_(e)DI +- n
+			// subl %%[re]bp,%%[re]di
```

Comment:
So why its added and subtracted?
Can't it use the in-place additions,
like (%rsi,%rbp,1)?
Why is there both rbp and ofs_MEMBASE?